.github/workflows/main.yml

```diff
@@ -57,7 +54,7 @@ jobs:
 
       - name: Build
         shell: bash -l {0}
-        run: make -j ${{ steps.cpu-cores.outputs.count }}
+        run: make -j ${{ runner.os == 'macOS' && 3 || 4 }}
```

Comment:
Why this change?

Reply:
Ah, looks like this GHA is now broken (launching too many compilations in parallel).

Reply:
It was because the `SimenB/github-actions-cpu-cores@v2` had not been working in a while. `macos-12` is the only one with 3 cores, all the others are 4 now.